nixos/modules/services/infrastructure/terraform-cloud-agent/default.nix

```diff
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.terraform-cloud-agent;
+in
+{
+  options.services.terraform-cloud-agent = {
+    enable = lib.mkEnableOption "terraform-cloud-agent";
+
+    name = lib.mkOption {
+      type = lib.types.str;
+    };
+
+    log-level = lib.mkOption {
+      type = lib.types.nullOr (lib.types.enum [ "trace" "debug" "info" "warn" "error" ]);
+      default = null;
+    };
+
+    data-dir = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+    };
+
+    cache-dir = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+    };
+
+    address = lib.mkOption {
+      type = lib.types.nullOr lib.types.str;
+      default = null;
+    };
+
+    tokenPath = lib.mkOption {
+      type = lib.types.path;
+      default = "/etc/terraform-cloud-agent-token";
+    };
+
+    accept = lib.mkOption {
+      type = lib.types.commas;
+      default = "plan,apply,policy,assessment";
+    };
```

Comment:
I recommend using a submodule for all the flags with a freeformType in addition to the known ones.

This is to make the module more future-proof to cli changes.
```suggestion
    tokenPath = lib.mkOption {
      type = lib.types.path;
      default = "/etc/terraform-cloud-agent-token";
    };

    data-dir = lib.mkOption {
      type = lib.types.nullOr lib.types.path;
      default = null;
    };

    cache-dir = lib.mkOption {
      type = lib.types.nullOr lib.types.path;
      default = null;
    };

    flags = lib.mkOption {
      freeformType = lib.types.attrsOf lib.types.str;

      name = lib.mkOption {
        type = lib.types.str;
      };

      log-level = lib.mkOption {
        type = lib.types.nullOr (lib.types.enum [ "trace" "debug" "info" "warn" "error" ]);
        default = null;
      };

      address = lib.mkOption {
        type = lib.types.nullOr lib.types.str;
        default = null;
      };

      accept = lib.mkOption {
        type = lib.types.commas;
        default = "plan,apply,policy,assessment";
      };

      auto-update = lib.mkOption {
        type = lib.types.str;
        readOnly = true;
        default = "disabled";
      };
    };
```

Reply:
TIL about freeformType. Thanks!

Addressed in 3d7a6140c9b4.